src/server/scripts/Outland/zone_blades_edge_mountains.cpp

```diff
@@ -33,75 +33,164 @@ EndContentData */
 enum deathsDoorWrapGate
 {
     SPELL_ARTILLERY_ON_THE_WRAP_GATE            = 39221,
-    SPELL_ANTI_DEMON_FLAME_THROWER              = 39222,
+    SPELL_IMP_AURA                              = 39227,
+    SPELL_HOUND_AURA                            = 39275,
+    SPELL_EXPLOSION                             = 30934,
+
+    NPC_EXPLOSION_BUNNY                         = 22502,
+    NPC_FEL_IMP                                 = 22474,
+    NPC_HOUND                                   = 22500,
     NPC_NORTH_GATE                              = 22471,
     NPC_SOUTH_GATE                              = 22472,
     NPC_NORTH_GATE_CREDIT                       = 22503,
     NPC_SOUTH_GATE_CREDIT                       = 22504,
 
+    GO_FIRE                                     = 185317,
+    GO_BIG_FIRE                                 = 185319
 };
 
 class npc_deahts_door_wrap_gate : public CreatureScript
 {
 public:
     npc_deahts_door_wrap_gate() : CreatureScript("npc_deahts_door_wrap_gate") { }
 
-    CreatureAI* GetAI(Creature* creature) const
-    {
-        return new npc_deahts_door_wrap_gateAI(creature);
-    }
-
     struct npc_deahts_door_wrap_gateAI : public ScriptedAI
     {
-        npc_deahts_door_wrap_gateAI(Creature *c) : ScriptedAI(c)
+        npc_deahts_door_wrap_gateAI(Creature* creature) : ScriptedAI(creature) { }
+
+        bool PartyTime;
+        uint64 PlayerGUID;
+        uint64 CannonGUID;
+        uint32 PartyTimer;
+        uint8 count;
+        float x, y, z, o;
+
+        void Reset() override
         {
+            PartyTime = false;
+            PlayerGUID = 0;
+            CannonGUID = 0;
+            PartyTimer = 0;
+            count = 0;
+            o = me->GetOrientation();
+            me->GetPosition(x, y, z);
+            x = x + 1.0f;
         }
 
-        uint8 count;
-        uint32 timer;
-        void Reset() { count = 0; timer = 0;}
-        void SpellHit(Unit* caster, SpellInfo const* spellInfo)
+        void SpellHit(Unit* caster, SpellInfo const* spell) override
         {
-            if (spellInfo->Id == SPELL_ARTILLERY_ON_THE_WRAP_GATE)
+            if (spell->Id == SPELL_ARTILLERY_ON_THE_WRAP_GATE)
             {
-                Player* plr = caster->GetCharmerOrOwnerPlayerOrPlayerItself();
-                if (!plr)
-                    return;
-
-                timer = 1;
                 count++;
 
+                if (count >= 1)
```

Comment:
The >= 1 is unnecessary since if you leave only if(count), any value that isn't 0 will pass this condition because any value that isn't 0 is true.